packages/superset-ui-connection/types/SupersetClient/index.d.ts

```diff
@@ -0,0 +1,76 @@
+type body = RequestInit['body'];
+type cache = RequestInit['cache'];
+type credentials = RequestInit['credentials'] | void;
+type csrfToken = string;
+type endpoint = string;
+type headers = { [key: string]: string };
+type host = string;
+type method = RequestInit['method'];
+type protocol = 'http:' | 'https:' | 'file:';
+type postPayload = { [key: string]: any };
+type mode = RequestInit['mode'];
+type redirect = RequestInit['redirect'];
+type timeout = number | void;
+type parseMethod = 'json' | 'text';
+type signal = RequestInit['signal'];
+type stringify = boolean;
+type url = string;
+
+interface ClientConfig {
```

Comment:
It seems like it's possible to further dedupe these interfaces a bit further, and pull out some types from `RequestInit` directly. Maybe something like: 
```typescript
interface ClientConfig extends Partial<Pick<RequestInit, 'credentials' | 'headers' | 'mode'>> {
  csrfToken?: csrfToken;
  host?: host;
  protocol?: protocol;
  timeout?: timeout;
}
```

Reply:
whoaaa had not heard of `Partial` or `Pick` 🤯